specviz/plugins/loader_wizard/loader_wizard.py

```diff
@@ -0,0 +1,631 @@
+            x = x.all_pix2world(np.arange(len(y)), 0)[0]
+
+        if x.shape != y.shape:
+            # TODO: status message
+            return
+
+        if yerr is not None and self.combo_uncertainty_type.currentText() == 'Inverse Variance':
+            yerr = np.sqrt(1 / yerr)
+
+        pen = pg.mkPen('k')
+
+        self.plot_widget.plot(x, y, pen=pen)
+        self.plot_widget.autoRange()
+
+        self.plot_widget.setLabels(left='{0} [{1}]'.format(self.helper_data.component,
+                                                           self.helper_data.unit),
+                                   bottom='{0} [{1}]'.format(self.helper_disp.component,
+                                                             self.helper_disp.unit))
+
+        if yerr is not None and yerr.shape == y.shape:
+            err = pg.ErrorBarItem(x=x, y=y, height=yerr, pen=pen)
+            self.plot_widget.addItem(err)
+
+    def _toggle_output_preview(self, event):
+        self.output_preview.set_text(self.as_new_loader())
+        self.output_preview.show()
+
+    def as_new_loader_dict(self):
+        raise NotImplementedError()
+
+    def get_template(self):
+        raise NotImplementedError()
+
+    def as_new_loader(self, name=None):
+
+        self.as_new_loader_dict(name=name)
+
+        template_path = self.get_template()
+
+        with open(template_path, 'r') as f:
+            filled_template = f.read().format(**self.new_loader_dict)
+
+        return filled_template
+
+    def _clear_loader_name_status(self):
+        self.ui.label_loader_name_status.setText('')
+        self.ui.label_loader_name_status.setStyleSheet('')
+
+    def save_loader_script(self, event=None):
+        if self.ui.loader_name.text() == "":
+            self.ui.label_loader_name_status.setText('Enter a name for the loader')
+            self.ui.label_loader_name_status.setStyleSheet('color: red')
+            return
+
+        specutils_dir = os.path.join(os.path.expanduser('~'), '.specutils')
+        if not os.path.exists(specutils_dir):
+            os.mkdir(specutils_dir)
+
+        filename = compat.getsavefilename(parent=self,
+                                          caption='Export loader to .py file',
+                                          basedir=specutils_dir)[0]
+
+        if filename == '':
+            return
+
+        filename = "{}.py".format(filename) if not filename.endswith(".py") else filename
+
+        string = self.as_new_loader()
+        with open(filename, 'w') as f:
+            f.write(string)
+
+        # If a loader by this name exists, delete it
+        if self.new_loader_dict['name'] in registry.get_formats()['Format']:
+            registry.unregister_reader(self.new_loader_dict['name'], Spectrum1D)
+
+        # Add new loader to registry
+        spec = importlib.util.spec_from_file_location(os.path.basename(filename)[:-3], filename)
+        mod = importlib.util.module_from_spec(spec)
+        spec.loader.exec_module(mod)
```

Comment:
Instead of these three lines, you can just do the following:

```python
specutils.io.registers._loader_user_io()
```

You'll need to add `import specutils` at the top of the module.